src/leds.cpp

```diff
@@ -0,0 +1,102 @@
+#include <unistd.h>
+#include <iostream>
+#include "leds.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "pigpio.h"
+
+
+
+led_flasher::led_flasher(){
+	std::cout << "construct led flasher" << std::endl;
+	decision = STAND;
+	return;
+};
+bool led_flasher::changeled(bool on){
+	if (on) {
+	   std::cout << "flash ON" << std::endl;
+	}
+	else {
+	   std::cout << "flash OFF" << std::endl;
+	}
+	return true;
+};
+bool led_flasher::mainthread(){
+	std::cout << "starting flasher" << std::endl;
+	int status;   
+	status = gpioInitialise();
+
+    if (status < 0)
+	{
+		fprintf(stderr, "pigpio initialisation failed.\n");
+		return 1;
+	}
+	bool done = false;
+	bool ledValue = false;
+	int i = 0;
+	while (i < 100) {
+		changeled(ledValue);
+		ledValue = !ledValue;
+		switch (decision) {
+			case HIT:
+				gpio(18, ledValue);
+				break;
+			case STAND:
+				gpio(22, ledValue);
+				break;
```

Comment:
Use MACROS to specify the pin numbers instead of raw/explicit numbers.


```suggestion
				gpio(HIT_PIN, ledValue);
				break;
			case STAND:
				gpio(STAND_PIN, ledValue);
				break;
```